www/includes/easyparliament/page.php

```diff
@@ -19,59 +19,65 @@ class PAGE {
-                // If the user tries to access the admin section when they're not
-                // allowed, then show them nothing.
-
-                if (!$THEUSER->isloggedin()) {
-                    $THISPAGE = new URL($this_page);
+    private function displayHeader() {
+        global $page_errors;
+        $h = new MySociety\TheyWorkForYou\Renderer\Header();
+        $u = new MySociety\TheyWorkForYou\Renderer\User();
 
-                    $LOGINURL = new URL('userlogin');
-                    $LOGINURL->insert(array('ret' => $THISPAGE->generate('none') ));
+        $data = $h->data;
+        $data = array_merge($u->data, $data);
+        if ( isset($page_errors) ) {
+            $data['page_errors'] = $page_errors;
+        }
+        $data['banner_text'] = '';
+        extract($data);
+        require_once INCLUDESPATH . 'easyparliament/templates/html/header.php';
 
-                    $text = "<a href=\"" . $LOGINURL->generate() . "\">You'd better sign in!</a>";
-                } else {
-                    $text = "That's all folks!";
-                }
+        echo '<div class="full-page legacy-page static-page"> <div class="full-page__row"> <div class="panel">';
 
-                $this_page = 'home';
+        $this->page_start_done = true;
+    }
 
-                $this->page_header();
-                $this->page_body();
-                $this->content_start();
-                $this->stripe_start();
+    private function checkForAdmin() {
+        global $DATA, $this_page, $THEUSER;
+        $parent = $DATA->page_metadata($this_page, "parent");
+        if ($parent == 'admin' && (!$THEUSER->isloggedin() || !$THEUSER->is_able_to('viewadminsection'))) {
+            if (!$THEUSER->isloggedin()) {
+                $THISPAGE = new URL($this_page);
 
-                print "<p>$text</p>\n";
+                $LOGINURL = new URL('userlogin');
+                $LOGINURL->insert(array('ret' => $THISPAGE->generate('none') ));
 
-                $this->stripe_end();
-                $this->page_end();
-                exit;
+                $text = "<a href=\"" . $LOGINURL->generate() . "\">You'd better sign in!</a>";
+            } else {
+                $text = "That's all folks!";
             }
-
-            $this->page_header();
-            $this->page_body();
-            $this->content_start();
-
-            $this->page_start_done = true;
-
+            $this_page = 'home';
+            $this->displayHeader();
+            echo $text;
+            $this->page_end();
+            exit();
         }
     }
 
+    public function page_end() {
+        if ( !$this->page_started() ) {
+            $this->page_start();
+        }
 
-    public function page_end($extra = null) {
-        $this->content_end();
-        $this->page_footer($extra);
+        echo '</div></div></div>';
+        $footer = new MySociety\TheyWorkForYou\Renderer\Footer();
+        $data['footer_links'] = $footer->data;
+        extract($data);
```

Comment:
`$data` isn't initialized, and then it's extracted, so I guess this could just be `$footer_links = $footer->data;`.